packages/client/src/components/App.tsx

```diff
@@ -1,37 +1,45 @@
-import React, {useEffect, useState} from "react";
-import {Route, Routes} from "react-router-dom";
-import {Shop} from "./Shop";
-import Cookies from "js-cookie";
-import {CartAdapter} from "../adapters/cart";
-import {Cart} from "./Cart";
-import {ProductCatalog} from "../adapters/productCatalog";
-import {OrderSummary} from "./OrderSummary";
-import {OrderAdapter} from "../adapters/order";
+import React from "react";
+import { Route, Routes } from "react-router-dom";
+import { Shop } from "./Shop";
+import { CartAdapter } from "../adapters/cart";
+import { Cart } from "./Cart";
+import { ProductCatalog } from "../adapters/productCatalog";
+import { OrderSummary } from "./OrderSummary";
+import { OrderAdapter } from "../adapters/order";
+import { useCartId } from "../hooks/useCartId";
 
 interface AppProps {
-    cartAdapter: CartAdapter,
-    catalog: ProductCatalog,
-    orderAdapter: OrderAdapter
-};
-
-export const App: React.FC<AppProps> = ({cartAdapter, catalog, orderAdapter}) => {
-
-    const [cartId, setCartId] = useState<string | null>(null);
-    useEffect(() => {
-        setCartId(new Date().getTime().toString());
-    }, []);
+  cartAdapter: CartAdapter;
+  catalog: ProductCatalog;
+  orderAdapter: OrderAdapter;
+}
 
-    useEffect(() => {
-        if (cartId) {
-            Cookies.set("cartId", cartId);
+export const App: React.FC<AppProps> = ({
+  cartAdapter,
+  catalog,
+  orderAdapter,
+}) => {
+  const cartId = useCartId();
```

Comment:
do you intentionally `useCartId` here and then push it to dependencies?

Reply:
Yes.
You can use it directly in each component since it maintains a fake internal state via localStorage, hence sort of a singleton.
This is to show the idea of custom hooks and non-dependent data sources.

Reply:
ok, it's actually better because it introduces fewer dependencies in the inner components